components/org.wso2.carbon.identity.api.server.workflow.engine/org.wso2.carbon.identity.rest.api.server.workflow.engine.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/workflow/engine/v1/core/WorkflowService.java

```diff
@@ -0,0 +1,631 @@
+                OptionDetails details = new OptionDetails();
+                details.setEntity(params[3]);
+                details.setValues(parameter.getParamValue());
+
+                // Ensure the list exists and add new details
+                if (templateParameters.getOptions() == null) {
+                    templateParameters.setOptions(new ArrayList<>());
+                }
+                templateParameters.getOptions().add(details);
+
+                // Update the map
+                templateParamsMap.put(stepNumber, templateParameters);
+            } else if ("HTSubject".equals(parameter.getParamName())){
+
+                if (parameter.getParamName().equals("HTSubject")){
+                    detailedWorkflow.setApprovalTask(parameter.getParamValue());
+                } else if ("HTDescription".equals(parameter.getParamName())){
+                    detailedWorkflow.setApprovalTaskDescription(parameter.getParamValue());
+                }
+            }
+        }
+        List<WorkflowTemplateParameters> templateParams = new ArrayList<>(templateParamsMap.values());
+
+        workflowTemplate.setProperties(templateParams);
+        detailedWorkflow.setWorkflowTemplate(workflowTemplate);
+        detailedWorkflow.setWorkflowEngine(workflowBean.getWorkflowImplId());
+
+        return detailedWorkflow;
+    }
+
+
+    /**
+     * List paginated workflows of a tenant.
+     *
+     * @param limit  Limit
+     * @param offset Offset
+     * @param filter filter
+     * @return WorkflowWizard[]
+     * @throws WorkflowException
+     */
+
+    public WorkflowSummary[] listPaginatedWorkflows(Integer limit, Integer offset, String filter) {
+
+        List<WorkflowSummary> workflowSummaryList = new ArrayList<>();
+        List<Workflow> workflowBeans = null;
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        try {
+            if (limit == null || offset == null){
+                limit = 15;
+                offset = 0;
+            } else {
+                limit = limit.intValue();
+                offset = offset.intValue();
+            }
+            //workflowBeans = WorkflowServiceDataHolder.getInstance().getWorkflowService().listPaginatedWorkflows(tenantId, limit, offset, filter);
+            workflowBeans = workflowManagementService.listPaginatedWorkflows(tenantId, limit, offset, filter);
+            for (Workflow workflow : workflowBeans) {
+                WorkflowSummary workflowTmp = getWorkflow(workflow);
+                workflowSummaryList.add(workflowTmp);
+            }
+            return workflowSummaryList.toArray(new WorkflowSummary[workflowSummaryList.size()]);
+
+        } catch (WorkflowException e) {
+            throw handleServerError(e.getErrorCode(),e.getMessage(), "", e);
+
+        }
+
+    }
+
+    /**
+     * Remove a workflow
+     *
+     * @param id ID of workflow to remove
+     * @return
+     * @throws WorkflowException
+     */
```

Comment:
Method comments should be descriptive.
Add a comment for the return type of. Refer other classes.